lib/utils/helpers.ts

```diff
@@ -202,6 +202,43 @@ function getApplication(
 	) as Promise<ApplicationWithDeviceType>;
 }
 
+/**
+ * Returns the boot partition number of the given image.
+ * @param imagePath Local filesystem path to a balenaOS image file
+ */
+export async function getBootPartition(imagePath: string): Promise<number> {
+	const imagefs = await import('balena-image-fs');
+	const filedisk = await import('file-disk');
+	const partitioninfo = await import('partitioninfo');
```

Comment:
I have a couple of questions regarding this change:

1. Would this change this pull request to a major change?
2. Should I edit the `package.json` file or the `npm-shrinkwrap.json` file to add the direct dependencies?

If the answer to the second question is the former, would it be simply just the addition of the two lines below?
```
"dependencies": {
    "file-disk": "^8.0.1",
    "partitioninfo": "^6.0.2",
    ...
  }
```

Thanks!

Reply:
Not a semver-major change, no. Dependencies are installed automatically through `npm install` and users themselves don't need to do anything in relation to the dependencies, so it doesn't affect patch/minor/major changes. And in this particular case, those dependencies were already being installed anyway because they were already dependencies of `balena-image-fs`.

Now that you mentioned it, I would normally describe a PR like this as a patch change (rather than minor), because we're really just fixing a bug, which is the fact that `os configure` and `local configure` were selecting the wrong partition when attempting to save some files in the image. A minor change implies a new feature, but from the balena CLI users' point of view, it's just a bug fix. By the way, regarding the first line of the commit message, which will automatically become a bullet point in the [CHANGELOG](https://github.com/balena-io/balena-cli/blob/master/CHANGELOG.md) file when the PR is merged:

* `Discover the boot partition number automatically when configuring system connections on an image.`

It is clear to you and me, but I think most CLI users would not readily understand how it affects them, not even users of the `local configure` command currently affected by the problem, because the `local configure` command does not have a `--system-connection` flag. So how about this for the first line of the commit message? -

* `os configure, local configure: Fix "Unsupported filesystem" error`

Then when reading the Changelog, users will understand that the new CLI release fixes that error for those commands. Users who have come across the error, or who are googling the error message ("unsupported filesystem") will readily understand that updating the CLI to that new version should fix the problem they were having with the `os configure` or `local configure` commands.

> would it be simply just the addition of the two lines below?

That's right, the changes are to `package.json` only, as you quoted. I gather that the `npm-shrinkwrap.json` file is not affected by this PR because those dependencies are already listed there.

Reply:
Thank you for the help.

I added the direct dependencies to the `package.json` file and also updated the first line of the commit message. 🏋🏼